util/src/main/java/io/kubernetes/client/util/ClientBuilder.java

```diff
@@ -12,231 +12,217 @@
+   * @throws IOException
+   *  if the configuration file or a file specified in a configuration file cannot be read.
+   */
+  public static ClientBuilder standard() throws IOException {
+    final FileReader kubeConfigReader = findConfigFromEnv();
+    if(kubeConfigReader != null) {
+      return kubeconfig(loadKubeConfig(kubeConfigReader));
+    }
+    final FileReader configReader = findConfigInHomeDir();
+    if(configReader != null) {
+      return kubeconfig(loadKubeConfig(configReader));
+    }
+    final File clusterCa = new File(SERVICEACCOUNT_CA_PATH);
+    if (clusterCa.exists()) {
+      return cluster();
+    }
+    return new ClientBuilder();
+  }
+
+  private static FileReader findConfigFromEnv() {
+    try {
+      String kubeConfig = System.getenv(ENV_KUBECONFIG);
+      if(kubeConfig == null) {
+        return null;
+      }
+      return new FileReader(kubeConfig);
+    } catch (FileNotFoundException e) {
+      log.info("Could not find file specified in $KUBECONFIG");
+      return null;
+    }
+  }
+
+  private static FileReader findConfigInHomeDir() {
+    try {
+      File config = new File(new File(System.getenv(ENV_HOME), KUBEDIR), KUBECONFIG);
+      return new FileReader(config);
+    } catch (FileNotFoundException e) {
+      log.info("Could not find ~/.kube/config");
+      return null;
+    }
+  }
+
+  /**
+   * Creates a builder which is pre-configured from the cluster configuration.
+   *
+   * @return <tt>ClientBuilder</tt> configured from the cluster configuration.
+   * @throws IOException
+   *  if the Service Account Token Path or CA Path is not readable.
+   */
+  public static ClientBuilder cluster() throws IOException {
+    final ClientBuilder builder = new ClientBuilder();
+
+    final String host = System.getenv(ENV_SERVICE_HOST);
+    final String port = System.getenv(ENV_SERVICE_PORT);
+    builder.setBasePath("https://" + host + ":" + port);
+
+    final String token = new String(Files.readAllBytes(Paths.get(SERVICEACCOUNT_TOKEN_PATH)),
+        Charset.defaultCharset());
+    builder.setCertificateAuthority(Files.readAllBytes(Paths.get(SERVICEACCOUNT_CA_PATH)));
+    builder.setAuthentication(new AccessTokenAuthentication(token));
+
+    return builder;
+  }
+
+  /**
+   * Creates a builder which is pre-configured from a {@link KubeConfig}.
+   *
+   * To load a <tt>KubeConfig</tt>, see {@link KubeConfig#loadKubeConfig(Reader)}.
+   *
+   * @param config
+   *  The {@link KubeConfig} to configure the builder from.
+   * @return <tt>ClientBuilder</tt> configured from the provided <tt>KubeConfig</tt>
+   * @throws IOException
+   *  if the files specified in the provided <tt>KubeConfig</tt> are not readable
+   */
+  public static ClientBuilder kubeconfig(KubeConfig config) throws IOException {
+    final ClientBuilder builder = new ClientBuilder();
+
+    String server = config.getServer();
+    if (!server.startsWith("http://") && !server.startsWith("https://")) {
```

Comment:
Maybe overkill, but just test for the presence of `"://"` ?

Reply:
Agreed. I wanted to preserve the original logic but will change to `.contains("://")`.